CoreFoundation/URL.subproj/CFURL.c

```diff
@@ -5208,3 +5208,152 @@ CFURLRef CFURLCreateFilePathURL(CFAllocatorRef alloc, CFURLRef url, CFErrorRef *
 
 CFURLRef CFURLCreateFileReferenceURL(CFAllocatorRef alloc, CFURLRef url, CFErrorRef *error) { return NULL; }
 
+// MARK: NSURL resource value functions
+
+#if __has_include(<linux/fs.h>)
+#include <sys/ioctl.h>
+#include <linux/fs.h>
+#endif
+
+#if DEPLOYMENT_TARGET_WINDOWS
+#include <windows.h>
+#endif
+
+Boolean __CFURLResourceValueHasFlag(CFURLRef url, CFURLResourcePropertyFlags key, errno_t *error) {
+    char pathBuf[CFMaxPathSize];
+    CFStringRef path = CFURLCopyPath(url);
+    if (!CFStringGetFileSystemRepresentation(path, pathBuf, CFMaxPathSize)) return false;
+    
+#if DEPLOYMENT_TARGET_MACOSX || DEPLOYMENT_TARGET_EMBEDDED || DEPLOYMENT_TARGET_FREEBSD
+    struct stat st;
+#endif
+    switch (key) {
+        case kCFURLResourceIsHidden:
+#if DEPLOYMENT_TARGET_MACOSX || DEPLOYMENT_TARGET_EMBEDDED || DEPLOYMENT_TARGET_FREEBSD
+            if (stat(pathBuf, &st)) {
+                error = errno;
+                return false;
+            }
+            return st.st_flags & UF_HIDDEN;
+#endif
+#if DEPLOYMENT_TARGET_WINDOWS
```

Comment:
```suggestion
#if TARGET_OS_WINDOWS
```

(I think you get the idea on these)